compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -1012,47 +877,23 @@ class AMDAIELowerToAIEPass
 
   AMDAIELowerToAIEPass() = default;
   AMDAIELowerToAIEPass(const AMDAIELowerToAIEPass &pass){};
-  void runOnOperation() override;
-};
-
-void AMDAIELowerToAIEPass::runOnOperation() {
-  // Main function call to convert all operations into AIE dialect operations
-  // inside an AIE device.
-  if (failed(lowerToAIE(getOperation()))) {
-    return signalPassFailure();
-  }
-  LLVM_DEBUG(llvm::dbgs() << "Module after lowerToAIE: " << getOperation());
-
-  // Clean up the HAL bindings and it's uses as they are not needed anymore.
-  if (failed(eraseHALBindings(getOperation()))) {
-    return signalPassFailure();
+  void runOnOperation() override {
+    // Main function call to convert all operations into AIE dialect
+    // operations inside an AIE device.
+    if (failed(lowerToAIE(getOperation()))) return signalPassFailure();
   }
-
-  // Move all dependencies, like for example constants, that are residing
-  // outside core operations into those core operations. This is required by
-  // the AIE dialect.
-  MLIRContext *context = &getContext();
-  RewritePatternSet patterns(context);
-  patterns.insert<MoveAllDependenciesIntoOp<xilinx::AIE::CoreOp>>(context);
-  if (failed(
-          applyPatternsAndFoldGreedily(getOperation(), std::move(patterns)))) {
-    return signalPassFailure();
-  }
-
-  // All Ukernel related function declarations will be within aie.device, so
-  // delete the ones outside from the SymbolTable.
-  SymbolTable symbolTable(getOperation());
-  getOperation()->walk([&](func::FuncOp funcOp) {
-    if (funcOp.isPrivate() && !funcOp->getParentOfType<AIE::DeviceOp>()) {
-      symbolTable.erase(funcOp);
-    }
-  });
-}
-
-}  // namespace
+};
 
 std::unique_ptr<Pass> createAMDAIELowerToAIEPass() {
   return std::make_unique<AMDAIELowerToAIEPass>();
 }
 
+void addAMDAIEToAIEPasses(OpPassManager &passManager) {
+  passManager.addPass(createAMDAIECanonicalizeNpuDmaCpyNdPass());
+  passManager.addPass(createCanonicalizerPass());
+  passManager.addPass(createAMDAIELowerToAIEPass());
+  passManager.addPass(createCanonicalizerPass());
+  passManager.addPass(createAMDAIESinkIntoCorePass());
+}
+
```

Comment:
I'm not sure if any other passes in IREE does this, but I guess it's not preferrable to add PassPipeline related constructs in a specific pass.

I'd add these within Passes.cpp itself.